msgpack-objc/MessagePack.m

```diff
@@ -354,11 +374,39 @@ case sizeof(uint64_t): {
             break;
         }
         default:
-            [NSException raise:NSInternalInconsistencyException format:@"Encountered unsupported TimeStamp format in MessagePack-data. Size is %li bytes but supported sizes are 32, 64 or 96 bits.", (unsigned long)self.data.length];
+            [NSException raise:NSInternalInconsistencyException format:@"Encountered unsupported TimeStamp format in MessagePack data. Size is %li bytes but supported sizes are 32, 64 or 96 bits.", (unsigned long)self.data.length];
     }
 
     NSTimeInterval interval = seconds + (nanoseconds / ONE_BILLION);
     return [NSDate dateWithTimeIntervalSince1970:interval];
 }
 
 @end
+
+@implementation DateExtension
+
+- (NSDate *)date
+{
+    if (self.data.length != 4) {
+        [NSException raise:NSInternalInconsistencyException format:@"Encountered unsupported Date format in MessagePack data. Size is %li bytes but only supported size is 4 bytes.", (unsigned long)self.data.length];
+    }
+    
+    uint8_t bytes[4];
+    [self.data getBytes:&bytes[0] length:self.data.length];
+    
+    uint16_t year = (bytes[0] << 8) | bytes[1];
```

Comment:
Is this easier to read maybe? Had to think for a while with the current line

(create a 16 bit pointer to the same data, take the first element)
```c
uint16_t year = *(uint16_t *)&bytes[0];
```

or alternatively, work with a pointer and take its value later:
```c
uint16_t *year = (uint16_t *)&bytes[0];
```

Reply:
True, that will look better.

Reply:
This does not work since the first byte is the one that is "worth" more. 
This works though `uint16_t data32 = CFSwapInt16BigToHost(*(uint32_t *)self.data.bytes);`

Reply:
Right, maybe keep it as it is then, we have to make sure it works on both simulator and device though because of different endianness!

Reply:
Or if you think your example is readable enough that should always work, so maybe it's nicer even though it's more text

Reply:
Ah ok, I will check what's happening on device :)

Reply:
Right. Well yeah, I used the `CFSwapInt16BigToHost` to solve that very problem. Wouldn't work on device otherwise. So I think it's good to use it here as well